homeassistant/components/influxdb/__init__.py

```diff
@@ -1,429 +1,61 @@
-    vol.Optional(CONF_BUCKET, default=DEFAULT_BUCKET): cv.string,
-}
-
-_CONFIG_SCHEMA_ENTRY = vol.Schema({vol.Optional(CONF_OVERRIDE_MEASUREMENT): cv.string})
-
-_CONFIG_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_EXCLUDE, default={}): vol.Schema(
-            {
-                vol.Optional(CONF_ENTITIES, default=[]): cv.entity_ids,
-                vol.Optional(CONF_DOMAINS, default=[]): vol.All(
-                    cv.ensure_list, [cv.string]
-                ),
-            }
-        ),
-        vol.Optional(CONF_INCLUDE, default={}): vol.Schema(
-            {
-                vol.Optional(CONF_ENTITIES, default=[]): cv.entity_ids,
-                vol.Optional(CONF_DOMAINS, default=[]): vol.All(
-                    cv.ensure_list, [cv.string]
-                ),
-            }
-        ),
-        vol.Optional(CONF_RETRY_COUNT, default=0): cv.positive_int,
-        vol.Optional(CONF_DEFAULT_MEASUREMENT): cv.string,
-        vol.Optional(CONF_OVERRIDE_MEASUREMENT): cv.string,
-        vol.Optional(CONF_TAGS, default={}): vol.Schema({cv.string: cv.string}),
-        vol.Optional(CONF_TAGS_ATTRIBUTES, default=[]): vol.All(
-            cv.ensure_list, [cv.string]
-        ),
-        vol.Optional(CONF_COMPONENT_CONFIG, default={}): vol.Schema(
-            {cv.entity_id: _CONFIG_SCHEMA_ENTRY}
-        ),
-        vol.Optional(CONF_COMPONENT_CONFIG_GLOB, default={}): vol.Schema(
-            {cv.string: _CONFIG_SCHEMA_ENTRY}
-        ),
-        vol.Optional(CONF_COMPONENT_CONFIG_DOMAIN, default={}): vol.Schema(
-            {cv.string: _CONFIG_SCHEMA_ENTRY}
-        ),
-    }
-)
-
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.All(
-            _CONFIG_SCHEMA.extend(COMPONENT_CONFIG_SCHEMA_CONNECTION),
-            validate_version_specific_config,
-            create_influx_url,
-        ),
-    },
-    extra=vol.ALLOW_EXTRA,
-)
-
-RE_DIGIT_TAIL = re.compile(r"^[^\.]*\d+\.?\d+[^\.]*$")
-RE_DECIMAL = re.compile(r"[^\d.]+")
-
-
-def get_influx_connection(client_kwargs, bucket):
-    """Create and check the correct influx connection for the API version."""
-    if bucket is not None:
-        # Test connection by synchronously writing nothing.
-        # If config is valid this will generate a `Bad Request` exception but not make anything.
-        # If config is invalid we will output an error.
-        # Hopefully a better way to test connection is added in the future.
-        try:
-            influx = InfluxDBClientV2(**client_kwargs)
-            influx.write_api(write_options=SYNCHRONOUS).write(bucket=bucket)
-
-        except ApiException as exc:
-            # 400 is the success state since it means we can write we just gave a bad point.
-            if exc.status != 400:
-                raise exc
-
-    else:
-        influx = InfluxDBClient(**client_kwargs)
-        influx.write_points([])
 
-    return influx
-
-
-def setup(hass, config):
+async def async_setup(hass, config):
```

Comment:
Why change to async api? The library doesn't support asyncio.

Reply:
I actually just thought that was the best practice now, I didn't realize you should be choosing async setup or not based on the library. I can switch it back.

Reply:
Since this integration doesn't use the config entries api yet, we don't need to switch to the async api and we try to avoid not needed context switches. So then usually it comes down to the library support if we should use the async api or not.

It would be ok to make this switch later as a preparation for config entries, if that makes sense.